lib/beaker/host/mac/pkg.rb

```diff
@@ -10,6 +10,25 @@ def install_package(name, cmdline_args = '', version = nil)
     execute("installer -pkg /Volumes/#{name}/#{name}.pkg -target /")
   end
 
+  # Install a package from a specified dmg
+  #
+  # @param [String] dmg_file      The dmg file, including path if not
+  #                               relative. Can be a URL.
+  # @param [String] pkg_base      The base name of the directory that the dmg
+  #                               attaches to under `/Volumes`
+  # @param [String] pkg_name      The name of the package file that should be
+  #                               used by the installer
+  # @example: Install vagrant from URL
+  #   mymachost.generic_install_dmg('https://releases.hashicorp.com/vagrant/1.8.4/vagrant_1.8.4.dmg', 'Vagrant', 'Vagrant.pkg')
+  def generic_install_dmg(dmg_file, pkg_base, pkg_name)
```

Comment:
can we refactor the [install_package](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/host/mac/pkg.rb#L8) method to use this one, so that we don't have the duplicate `hdiutil` and `installer` commands?